src/Core/Analysis/IAnalysisIssue.cs

```diff
@@ -39,6 +41,23 @@ public interface IAnalysisIssue
         string Message { get; }
 
         string FilePath { get; }
+
+        IReadOnlyList<IAnalysisIssueLocation> Locations { get; }
+    }
+
+    public interface IAnalysisIssueLocation
```

Comment:
I'm debating if IAnalysisIssue should inherit from IAnalysisIssueLocation, as the "main" location is a location as well. WDYT?

Reply:
No, I don't think so - doesn't feel right, somehow. "Favour composition over inheritance" - if it needs to have a "primary location", give it a location property.